mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1829,9 +1845,37 @@ impl TakerSwap {
             tx_hash,
         };
 
-        Ok((Some(TakerSwapCommand::Finish), vec![TakerSwapEvent::MakerPaymentSpent(
-            tx_ident,
-        )]))
+        Ok((Some(TakerSwapCommand::ConfirmMakerPaymentSpend), vec![
+            TakerSwapEvent::MakerPaymentSpent(tx_ident),
+        ]))
+    }
+
+    async fn confirm_maker_payment_spend(&self) -> Result<(Option<TakerSwapCommand>, Vec<TakerSwapEvent>), String> {
+        // we should wait for only one confirmation to make sure our spend transaction is not failed
+        let confirm_maker_payment_spend_input = ConfirmPaymentInput {
+            payment_tx: self.r().maker_payment_spend.clone().unwrap().tx_hex.0,
+            confirmations: std::cmp::min(1, self.r().data.maker_payment_confirmations),
```

Comment:
Can maker_payment_confirmations be 0? 
Should 'confirmations' be also 0 in this case?

Reply:
> Can maker_payment_confirmations be 0?

seems like yes. I took `confirm_taker_payment_spend` from maker_swap.rs as an example
https://github.com/KomodoPlatform/komodo-defi-framework/blob/0073fe80d5d89fbd055fa96c6199f73139b3dc9e/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L1143-L1153

std::cmp::min ensures that at least one confirmation is required unless maker_payment_confirmations is explicitly set to a value less than 1.

> Should 'confirmations' be also 0 in this case?

Previously confirmations value always was 1, but was changed in this PR

https://github.com/KomodoPlatform/komodo-defi-framework/pull/971/files#diff-b837d781bf1a7f0fb7828ac91d7b96f2f75328bb14f5cd47c2dd316f837a2574R796-R797

@shamardy could you clarify please, why return minimum between 1 and the value of taker_payment_confirmations was added previously? From what I see, we decided to allow to skip payment spend confirmation step?

Reply:
Yes, we allow the user to use 0 confirmations if they want since 0 confirmations will be used in a more important step before that if it was set https://github.com/KomodoPlatform/komodo-defi-framework/blob/1461212c1e1807b0031a78e1b000d1d2ff0491ed/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L982-L997

Reply:
Ok, I will update `we should wait for only one confirmation...` comment then

Reply:
added note https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/commits/f93fcde3244513442856bbfc210f78853e7f1f4a